### ISSUE TITLE
Issues with the layout core/gallery block because margin top on first element

### ISSUE THREAD

**Issue by fabiankaegy** [WordPress/twentynineteen#58]
<img width="545" alt="bildschirmfoto 2018-10-16 um 21 45 42" src="https://user-images.githubusercontent.com/20684594/47043068-fae8fd80-d18c-11e8-8fa0-ee9abe01fe49.png">
<img width="1097" alt="bildschirmfoto 2018-10-16 um 22 15 54" src="https://user-images.githubusercontent.com/20684594/47044664-135b1700-d191-11e8-91cb-bf0e548b9759.png">

**Comment by fabiankaegy** [comment]
The visual jump on the first image in the gallery is caused by line 30 in `sass/blocks/_blocks.scss`.
```sass 
.entry-content > * > *:first-child, 
.entry-summary > * > *:first-child {
    margin-top: 0;
}
```

**Comment by fabiankaegy** [comment]
It also applies to `:last-child` `margin-bottom: 0;`

**Comment by fabiankaegy** [comment]
<img width="999" alt="bildschirmfoto 2018-10-16 um 22 39 12" src="https://user-images.githubusercontent.com/20684594/47045948-5b2f6d80-d194-11e8-8f0f-88526140a8de.png">

**Comment by joyously** [comment]
There was discussion in #core-editor about how to fix margin collapsing in general.

**Comment by fabiankaegy** [comment]
This also seems to be fixed in Gutenberg 4.0 
<img width="504" alt="bildschirmfoto 2018-10-18 um 09 18 32" src="https://user-images.githubusercontent.com/20684594/47138711-709eb780-d2ba-11e8-82c4-c3e79850ccb8.png">
